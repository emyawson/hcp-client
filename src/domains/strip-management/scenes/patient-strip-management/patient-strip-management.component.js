import React from 'react';

import {
  Tabs,
  Card,
  CardMinimizer,
  GridContainer,
  GridItem,
  TabLinkItem,
  TabsHeader,
  TabsContent,
} from 'src/components';
import { PatientSummaryBarContainer } from 'src/widgets/patient-summary-bar';
import { WithPermissions } from 'src/utils/with-permissions/with-permissions.container';
import { PERMISSIONS } from 'src/core/permissions/permissions.constants';
import { translate } from 'src/i18n';

export const PatientStripManagementComp = ({ match, patient, children }) => (
  <GridContainer marginBottom>
    <GridItem span="12">
      <PatientSummaryBarContainer />
    </GridItem>
    <GridItem span="12">
      <Card>
        <WithPermissions hasPermissions={[PERMISSIONS.PATIENT_DIAGNOSTICS]}>
          <CardMinimizer link={`/patients/${match.params.id}`} />
        </WithPermissions>
        <Tabs>
          <TabsHeader mb={4} mr="auto" ml="0">
            <TabLinkItem
              exact
              path=""
              name={translate('stripDelivery.title')}
            />
            <WithPermissions hasPermissions={[PERMISSIONS.STRIP_PRESCRIPTIONS]}>
              <TabLinkItem
                path="prescription"
                name={translate('prescription.title')}
              />
            </WithPermissions>
            <WithPermissions
              hasPermissions={[PERMISSIONS.STRIP_PATIENT_ALERTS]}
            >
              <TabLinkItem
                exact
                path="alerts"
                name={translate('alerts.title')}
              />
            </WithPermissions>
          </TabsHeader>
          <TabsContent>{children}</TabsContent>
        </Tabs>
      </Card>
    </GridItem>
  </GridContainer>
);
