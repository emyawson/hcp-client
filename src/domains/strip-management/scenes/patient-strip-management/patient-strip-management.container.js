import { compose, lifecycle } from 'recompose';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { isEmpty } from 'ramda';

import { mapDispatchers } from 'src/utils';
import { getDeliveryRequest, fetchPatientRequest } from 'src/core';
import { withNavigators } from 'src/utils/with-navigators';
import { getPatientStockRequest } from 'src/core';
import {
  getCurrentPrescriptionRequest,
  getStripModelsRequest,
} from 'src/core/prescription';

import { PatientStripManagementComp } from './patient-strip-management.component';
import { patientStripManagementConnector } from './store/patient-strip-management.selector';

const dispatchers = mapDispatchers({
  getPatientStockRequest: getPatientStockRequest.start,
  onFetchPatient: fetchPatientRequest.start,
  getDeliveryRequest: getDeliveryRequest.start,
  getCurrentPrescriptionRequest: getCurrentPrescriptionRequest.start,
  getStripModelsRequest: getStripModelsRequest.start,
});

export const addPatientStripManagement = Component =>
  compose(
    connect(
      patientStripManagementConnector,
      dispatchers,
    ),
    lifecycle({
      componentDidMount() {
        const patientId = this.props.match.params.id;
        const { clinicId, departmentId, stripModels } = this.props;

        this.props.onFetchPatient({ patientId });
        this.props.getDeliveryRequest({ patientId });
        this.props.getCurrentPrescriptionRequest({ patientId });

        if (isEmpty(stripModels)) {
          this.props.getStripModelsRequest({ clinicId, departmentId });
        }
      },
    }),
  )(Component);

export const PatientStripManagementWrapper = compose(
  withNavigators({
    hasLeftNav: true,
    hasTopNav: true,
  }),
  withRouter,
  addPatientStripManagement,
)(PatientStripManagementComp);
