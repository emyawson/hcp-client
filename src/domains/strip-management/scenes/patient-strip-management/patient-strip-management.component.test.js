import React from 'react';
import { shallow } from 'enzyme';

import { PatientStripManagementComp } from './patient-strip-management.component';

describe('Patient Strip Management component', () => {
  it('renders correctly', () => {
    const props = {
      match: {
        params: {
          id: 22,
        },
      },
      patient: {
        name: 'Patient Test',
      },
    };
    const component = shallow(<PatientStripManagementComp {...props} />);
    expect(component).toMatchSnapshot();
  });
});
