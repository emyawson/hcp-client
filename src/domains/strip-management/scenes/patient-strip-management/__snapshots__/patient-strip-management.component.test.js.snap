// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient Strip Management component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    match={
      Object {
        "params": Object {
          "id": 22,
        },
      }
    }
    patient={
      Object {
        "name": "Patient Test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <GridItem
          span="12"
        >
          <Connect(withRouter(lifecycle(Component))) />
        </GridItem>,
        <GridItem
          span="12"
        >
          <Card>
            <Connect(mapProps(WithPermissionsComponent))
              hasPermissions={
                Array [
                  "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                ]
              }
            >
              <Unknown
                link="/patients/22"
              />
            </Connect(mapProps(WithPermissionsComponent))>
            <Unknown>
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  exact={true}
                  name="Strip Delivery"
                  path=""
                />
                <Connect(mapProps(WithPermissionsComponent))
                  hasPermissions={
                    Array [
                      "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                    ]
                  }
                >
                  <withRouter()
                    name="GUIDELINE"
                    path="prescription"
                  />
                </Connect(mapProps(WithPermissionsComponent))>
                <Connect(mapProps(WithPermissionsComponent))
                  hasPermissions={
                    Array [
                      "ROLE_STRIP_MNG_ALERTS_CONFIG",
                    ]
                  }
                >
                  <withRouter()
                    exact={true}
                    name="Alerts"
                    path="alerts"
                  />
                </Connect(mapProps(WithPermissionsComponent))>
              </TabsHeader>
              <Unknown />
            </Unknown>
          </Card>
        </GridItem>,
      ],
      "marginBottom": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(withRouter(lifecycle(Component))) />,
          "span": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Card>
            <Connect(mapProps(WithPermissionsComponent))
              hasPermissions={
                Array [
                  "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                ]
              }
            >
              <Unknown
                link="/patients/22"
              />
            </Connect(mapProps(WithPermissionsComponent))>
            <Unknown>
              <TabsHeader
                mb={4}
                ml="0"
                mr="auto"
              >
                <withRouter()
                  exact={true}
                  name="Strip Delivery"
                  path=""
                />
                <Connect(mapProps(WithPermissionsComponent))
                  hasPermissions={
                    Array [
                      "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                    ]
                  }
                >
                  <withRouter()
                    name="GUIDELINE"
                    path="prescription"
                  />
                </Connect(mapProps(WithPermissionsComponent))>
                <Connect(mapProps(WithPermissionsComponent))
                  hasPermissions={
                    Array [
                      "ROLE_STRIP_MNG_ALERTS_CONFIG",
                    ]
                  }
                >
                  <withRouter()
                    exact={true}
                    name="Alerts"
                    path="alerts"
                  />
                </Connect(mapProps(WithPermissionsComponent))>
              </TabsHeader>
              <Unknown />
            </Unknown>
          </Card>,
          "span": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Connect(mapProps(WithPermissionsComponent))
                hasPermissions={
                  Array [
                    "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                  ]
                }
              >
                <Unknown
                  link="/patients/22"
                />
              </Connect(mapProps(WithPermissionsComponent))>,
              <Unknown>
                <TabsHeader
                  mb={4}
                  ml="0"
                  mr="auto"
                >
                  <withRouter()
                    exact={true}
                    name="Strip Delivery"
                    path=""
                  />
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                      ]
                    }
                  >
                    <withRouter()
                      name="GUIDELINE"
                      path="prescription"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_ALERTS_CONFIG",
                      ]
                    }
                  >
                    <withRouter()
                      exact={true}
                      name="Alerts"
                      path="alerts"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                </TabsHeader>
                <Unknown />
              </Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown
                  link="/patients/22"
                />,
                "hasPermissions": Array [
                  "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "link": "/patients/22",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <TabsHeader
                    mb={4}
                    ml="0"
                    mr="auto"
                  >
                    <withRouter()
                      exact={true}
                      name="Strip Delivery"
                      path=""
                    />
                    <Connect(mapProps(WithPermissionsComponent))
                      hasPermissions={
                        Array [
                          "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                        ]
                      }
                    >
                      <withRouter()
                        name="GUIDELINE"
                        path="prescription"
                      />
                    </Connect(mapProps(WithPermissionsComponent))>
                    <Connect(mapProps(WithPermissionsComponent))
                      hasPermissions={
                        Array [
                          "ROLE_STRIP_MNG_ALERTS_CONFIG",
                        ]
                      }
                    >
                      <withRouter()
                        exact={true}
                        name="Alerts"
                        path="alerts"
                      />
                    </Connect(mapProps(WithPermissionsComponent))>
                  </TabsHeader>,
                  <Unknown />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <withRouter()
                        exact={true}
                        name="Strip Delivery"
                        path=""
                      />,
                      <Connect(mapProps(WithPermissionsComponent))
                        hasPermissions={
                          Array [
                            "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                          ]
                        }
                      >
                        <withRouter()
                          name="GUIDELINE"
                          path="prescription"
                        />
                      </Connect(mapProps(WithPermissionsComponent))>,
                      <Connect(mapProps(WithPermissionsComponent))
                        hasPermissions={
                          Array [
                            "ROLE_STRIP_MNG_ALERTS_CONFIG",
                          ]
                        }
                      >
                        <withRouter()
                          exact={true}
                          name="Alerts"
                          path="alerts"
                        />
                      </Connect(mapProps(WithPermissionsComponent))>,
                    ],
                    "mb": 4,
                    "ml": "0",
                    "mr": "auto",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "exact": true,
                        "name": "Strip Delivery",
                        "path": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": <withRouter()
                          name="GUIDELINE"
                          path="prescription"
                        />,
                        "hasPermissions": Array [
                          "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                        ],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "function",
                        "props": Object {
                          "name": "GUIDELINE",
                          "path": "prescription",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": <withRouter()
                          exact={true}
                          name="Alerts"
                          path="alerts"
                        />,
                        "hasPermissions": Array [
                          "ROLE_STRIP_MNG_ALERTS_CONFIG",
                        ],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "function",
                        "props": Object {
                          "exact": true,
                          "name": "Alerts",
                          "path": "alerts",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <GridItem
            span="12"
          >
            <Connect(withRouter(lifecycle(Component))) />
          </GridItem>,
          <GridItem
            span="12"
          >
            <Card>
              <Connect(mapProps(WithPermissionsComponent))
                hasPermissions={
                  Array [
                    "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                  ]
                }
              >
                <Unknown
                  link="/patients/22"
                />
              </Connect(mapProps(WithPermissionsComponent))>
              <Unknown>
                <TabsHeader
                  mb={4}
                  ml="0"
                  mr="auto"
                >
                  <withRouter()
                    exact={true}
                    name="Strip Delivery"
                    path=""
                  />
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                      ]
                    }
                  >
                    <withRouter()
                      name="GUIDELINE"
                      path="prescription"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_ALERTS_CONFIG",
                      ]
                    }
                  >
                    <withRouter()
                      exact={true}
                      name="Alerts"
                      path="alerts"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                </TabsHeader>
                <Unknown />
              </Unknown>
            </Card>
          </GridItem>,
        ],
        "marginBottom": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(withRouter(lifecycle(Component))) />,
            "span": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Card>
              <Connect(mapProps(WithPermissionsComponent))
                hasPermissions={
                  Array [
                    "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                  ]
                }
              >
                <Unknown
                  link="/patients/22"
                />
              </Connect(mapProps(WithPermissionsComponent))>
              <Unknown>
                <TabsHeader
                  mb={4}
                  ml="0"
                  mr="auto"
                >
                  <withRouter()
                    exact={true}
                    name="Strip Delivery"
                    path=""
                  />
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                      ]
                    }
                  >
                    <withRouter()
                      name="GUIDELINE"
                      path="prescription"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                  <Connect(mapProps(WithPermissionsComponent))
                    hasPermissions={
                      Array [
                        "ROLE_STRIP_MNG_ALERTS_CONFIG",
                      ]
                    }
                  >
                    <withRouter()
                      exact={true}
                      name="Alerts"
                      path="alerts"
                    />
                  </Connect(mapProps(WithPermissionsComponent))>
                </TabsHeader>
                <Unknown />
              </Unknown>
            </Card>,
            "span": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Connect(mapProps(WithPermissionsComponent))
                  hasPermissions={
                    Array [
                      "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                    ]
                  }
                >
                  <Unknown
                    link="/patients/22"
                  />
                </Connect(mapProps(WithPermissionsComponent))>,
                <Unknown>
                  <TabsHeader
                    mb={4}
                    ml="0"
                    mr="auto"
                  >
                    <withRouter()
                      exact={true}
                      name="Strip Delivery"
                      path=""
                    />
                    <Connect(mapProps(WithPermissionsComponent))
                      hasPermissions={
                        Array [
                          "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                        ]
                      }
                    >
                      <withRouter()
                        name="GUIDELINE"
                        path="prescription"
                      />
                    </Connect(mapProps(WithPermissionsComponent))>
                    <Connect(mapProps(WithPermissionsComponent))
                      hasPermissions={
                        Array [
                          "ROLE_STRIP_MNG_ALERTS_CONFIG",
                        ]
                      }
                    >
                      <withRouter()
                        exact={true}
                        name="Alerts"
                        path="alerts"
                      />
                    </Connect(mapProps(WithPermissionsComponent))>
                  </TabsHeader>
                  <Unknown />
                </Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    link="/patients/22"
                  />,
                  "hasPermissions": Array [
                    "ROLE_USER_SHOW_GRAPHIC_CONFIGURATION",
                  ],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "link": "/patients/22",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <TabsHeader
                      mb={4}
                      ml="0"
                      mr="auto"
                    >
                      <withRouter()
                        exact={true}
                        name="Strip Delivery"
                        path=""
                      />
                      <Connect(mapProps(WithPermissionsComponent))
                        hasPermissions={
                          Array [
                            "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                          ]
                        }
                      >
                        <withRouter()
                          name="GUIDELINE"
                          path="prescription"
                        />
                      </Connect(mapProps(WithPermissionsComponent))>
                      <Connect(mapProps(WithPermissionsComponent))
                        hasPermissions={
                          Array [
                            "ROLE_STRIP_MNG_ALERTS_CONFIG",
                          ]
                        }
                      >
                        <withRouter()
                          exact={true}
                          name="Alerts"
                          path="alerts"
                        />
                      </Connect(mapProps(WithPermissionsComponent))>
                    </TabsHeader>,
                    <Unknown />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <withRouter()
                          exact={true}
                          name="Strip Delivery"
                          path=""
                        />,
                        <Connect(mapProps(WithPermissionsComponent))
                          hasPermissions={
                            Array [
                              "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                            ]
                          }
                        >
                          <withRouter()
                            name="GUIDELINE"
                            path="prescription"
                          />
                        </Connect(mapProps(WithPermissionsComponent))>,
                        <Connect(mapProps(WithPermissionsComponent))
                          hasPermissions={
                            Array [
                              "ROLE_STRIP_MNG_ALERTS_CONFIG",
                            ]
                          }
                        >
                          <withRouter()
                            exact={true}
                            name="Alerts"
                            path="alerts"
                          />
                        </Connect(mapProps(WithPermissionsComponent))>,
                      ],
                      "mb": 4,
                      "ml": "0",
                      "mr": "auto",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "function",
                        "props": Object {
                          "exact": true,
                          "name": "Strip Delivery",
                          "path": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "class",
                        "props": Object {
                          "children": <withRouter()
                            name="GUIDELINE"
                            path="prescription"
                          />,
                          "hasPermissions": Array [
                            "ROLE_STRIP_MNG_STRIP_PRESCRIPTION",
                          ],
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "function",
                          "props": Object {
                            "name": "GUIDELINE",
                            "path": "prescription",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "class",
                        "props": Object {
                          "children": <withRouter()
                            exact={true}
                            name="Alerts"
                            path="alerts"
                          />,
                          "hasPermissions": Array [
                            "ROLE_STRIP_MNG_ALERTS_CONFIG",
                          ],
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "function",
                          "props": Object {
                            "exact": true,
                            "name": "Alerts",
                            "path": "alerts",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
